app/policies/subject_rule_effect_policy.rb

```diff
@@ -38,26 +21,38 @@ def edit?
   end
 
   def update?
-    # record is a subject_rule_effect from the controller
-    if credential.admin?
-      true
-    else
-      # this is a good place to check the defined subect set id in the
-      # rule effect belongs to the scoped credential project ids using the API client
+    return true if credential.admin?
+    return false unless valid_credentials?
```

Comment:
Swap these, in case the admin credentials is not valid as well (can't see how but...)
```suggestion
    return false unless valid_credentials?
    return true if credential.admin?
```